### ISSUE TITLE
BCELoss and BCEWithLogitsLoss differ when one of the input logits is float("inf")

### ISSUE THREAD

**Issue by dhruvbird** [pytorch/pytorch#102894]
### 🐛 Describe the bug

Based on the docs, it seems like `BCEWithLogitsLoss` is supposed to be the more numerically table sibling of `BCELoss`. However, when one of the input logits to `BCEWithLogitsLoss` is `float("inf")` the result is `Nan` whereas `sigmoid(inf) = 1.0`, so it should probably be well defined. When I use `BCELoss(sigmoid(x), target)` it gives the desired result.

```
x = torch.tensor([float("inf"), 1.0])
y = torch.tensor([1.0, 0.0])

loss = F.binary_cross_entropy_with_logits(x, y)
print (loss)

loss = F.binary_cross_entropy(x.sigmoid(), y)
print (loss)

print(torch.nn.BCEWithLogitsLoss()(x, y))
print(torch.nn.BCELoss()(x.sigmoid(), y))
```

Output:

```
tensor(nan)
tensor(0.6566)
tensor(nan)
tensor(0.6566)
```

### Versions

1.12.1

**Comment by tushar-thoriya** [comment]
can i take this ?

**Comment by janeyx99** [comment]
Yes! Feel free to send a PR fixing this discrepancy with a test case.

**Comment by cleonard530** [comment]
It seems @tushar-thoriya hasn't worked on this in a while so I will take a look at it.
